### PR TITLE
Replace usages of const std::vector<T>& in parameters

### PR DESCRIPTION
**Pull request by beckerhe**
Replace usages of const std::vector<T>& in parameters
    
This replaces the usage of `const std::vector<T>&` in function
parameters by `absl::Span<T const>`.
    
Using Span has the advantage that the caller can pass in any consecutive range
of elements - a std::array for example. It also makes potentially
expensive copies more explicit. More details here: http://go/totw/93

There are two commits. The first one only contains mechanical changes (done by clang-tidy).
All the less trivial manual changes are in the second commit. When reviewing you probably want
to review the second commit more thoroughly than the first one.

### CODE REVIEW

**Review by akopich — APPROVED** [review]
Thanks for the nice change! 

My only concern is `T const` vs `const T`, as we seem to use the latter everywhere else.

**Comment by beckerhe** [comment]
> Thanks for the nice change!
> 
> My only concern is `T const` vs `const T`, as we seem to use the latter everywhere else.

Yeah. Unfortunately this is how it comes out of clang-tidy. (The `const` needs to at the end to apply correctly to pointer-like types. Ideally clang-format should clean this up. I will look if this can be activated.

**Comment by beckerhe** [comment]
> > Thanks for the nice change!
> > My only concern is `T const` vs `const T`, as we seem to use the latter everywhere else.
> 
> Yeah. Unfortunately this is how it comes out of clang-tidy. (The `const` needs to at the end to apply correctly to pointer-like types. Ideally clang-format should clean this up. I will look if this can be activated.

Well, it kinda works. :grin: 

```diff
       .WillByDefault([&capture_events_received_count](
-                         absl::Span<orbit_grpc_protos::ProducerCaptureEvent const> events) {
+                         absl::Span<orbit_grpc_protos::const ProducerCaptureEvent> events) {
         capture_events_received_count += events.size();
```

I guess the feature still has some bugs in LLVM-14 (where it was introduced). I will try a newer version of clang format.

File: src/CaptureEventProducer/LockFreeBufferCaptureEventProducerTest.cpp [hunk 1]
```diff
@@ -95,7 +96,7 @@ TEST_F(LockFreeBufferCaptureEventProducerTest, EnqueueIntermediateEventIfCapturi
   std::atomic<uint64_t> capture_events_received_count = 0;
   ON_CALL(*fake_service_, OnCaptureEventsReceived)
       .WillByDefault([&capture_events_received_count](
-                         const std::vector<orbit_grpc_protos::ProducerCaptureEvent>& events) {
+                         absl::Span<orbit_grpc_protos::ProducerCaptureEvent const> events) {
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/CaptureEventProducer/LockFreeBufferCaptureEventProducerTest.cpp:99]: That would be indeed pretty nice.

**Review by florian-kuebler — APPROVED** [review]
I agree with @akopich on the Span<T const>. But looks good. Thanks!

File: src/CaptureEventProducer/LockFreeBufferCaptureEventProducerTest.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (beckerhe) [reply src/CaptureEventProducer/LockFreeBufferCaptureEventProducerTest.cpp]: I took what clang format suggested and fixed it up manually. PTAL!

**Comment by beckerhe** [comment]
> > > Thanks for the nice change!
> > > My only concern is `T const` vs `const T`, as we seem to use the latter everywhere else.
> > 
> > 
> > Yeah. Unfortunately this is how it comes out of clang-tidy. (The `const` needs to at the end to apply correctly to pointer-like types. Ideally clang-format should clean this up. I will look if this can be activated.
> 
> Well, it kinda works. grin
> 
> ```diff
>        .WillByDefault([&capture_events_received_count](
> -                         absl::Span<orbit_grpc_protos::ProducerCaptureEvent const> events) {
> +                         absl::Span<orbit_grpc_protos::const ProducerCaptureEvent> events) {
>          capture_events_received_count += events.size();
> ```
> 
> I guess the feature still has some bugs in LLVM-14 (where it was introduced). I will try a newer version of clang format.

For the sake of completeness, this is still broken in LLVM-15 (The latest release and also the latest version I had at hand.)

**Pull request merged**